editoast/src/views/infra/edition.rs

```diff
@@ -54,6 +80,753 @@ pub async fn edit<'a>(
+        .bind::<BigInt, _>(infra_id)
+        .bind::<Text, _>(payload.id.to_string())
+        .bind::<Double, _>(distance_fraction)
+        .load(&mut conn)
+        .await?;
+    let tracksection_data = result[0].clone();
+    let tracksection = tracksection_data.railjson.as_ref().clone();
+
+    // Building the two newly tracksections from the splitted one
+    // ~~~~~~~~~~~~~~~
+    // left
+    let left_tracksection_id = Uuid::new_v4();
+    let left_tracksection = TrackSection {
+        id: Identifier::from(left_tracksection_id),
+        length: distance,
+        geo: tracksection_data.left_geo.as_ref().clone(),
+        sch: tracksection_data.left_geo.as_ref().clone(),
+        slopes: tracksection
+            .slopes
+            .iter()
+            .filter(|e| e.begin <= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.end > distance {
+                    item.end = distance;
+                }
+                return item;
+            })
+            .collect_vec(),
+        curves: tracksection
+            .curves
+            .iter()
+            .filter(|e| e.begin <= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.end > distance {
+                    item.end = distance;
+                }
+                return item;
+            })
+            .collect_vec(),
+        loading_gauge_limits: tracksection
+            .loading_gauge_limits
+            .iter()
+            .filter(|e| e.begin <= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.end > distance {
+                    item.end = distance;
+                }
+                return item;
+            })
+            .collect_vec(),
+        ..tracksection.clone()
+    };
+    // right
+    let right_tracksection_id = Uuid::new_v4();
+    let right_tracksection = TrackSection {
+        id: Identifier::from(right_tracksection_id),
+        length: tracksection.length - distance,
+        geo: tracksection_data.right_geo.as_ref().clone(),
+        sch: tracksection_data.right_geo.as_ref().clone(),
+        slopes: tracksection
+            .slopes
+            .iter()
+            .filter(|e| e.end >= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.begin < distance {
+                    item.begin = distance;
+                } else {
+                    item.begin = item.begin - distance;
+                }
+                item.end = item.end - distance;
+                return item;
+            })
+            .collect_vec(),
+        curves: tracksection
+            .curves
+            .iter()
+            .filter(|e| e.end >= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.begin < distance {
+                    item.begin = distance;
```

Comment:
This has to be fixed everywhere
```suggestion
                    item.begin = 0.;
```